src/platform/packages/shared/dashboards/dashboards-selector/components/dashboards_selector.tsx

```diff
@@ -41,50 +82,26 @@ export function DashboardsSelector({
   const [isComboBoxOpen, setIsComboBoxOpen] = useState(false);
 
   const fetchDashboardTitles = useCallback(async () => {
-    if (!dashboardsFormData?.length || !contentManagement) {
+    if (!dashboardsFormData?.length) {
       return;
     }
 
     try {
-      const dashboardPromises = dashboardsFormData.map(async (dashboard) => {
-        try {
-          const fetchedDashboard = await dashboardServiceProvider(contentManagement).fetchDashboard(
-            dashboard.id
-          );
-
-          // Only return the dashboard if it exists, fetch was successful, and has a title
-          if (
-            fetchedDashboard &&
-            fetchedDashboard.status === 'success' &&
-            fetchedDashboard.attributes?.title
-          ) {
-            return {
-              label: fetchedDashboard.attributes.title,
-              value: dashboard.id,
-            };
-          }
-          // Return null if dashboard doesn't have required data
-          return null;
-        } catch (dashboardError) {
-          /**
-           * Swallow the error that is thrown, since this just means the selected dashboard was deleted
-           * Return null when dashboard fetch fails
-           */
-          return null;
-        }
-      });
-
-      const results = await Promise.all(dashboardPromises);
+      // Fetch dashboards by their IDs directly instead of fetching all dashboards
```

Comment:
not sure comment is needed since it describes a change that no longer exists in the code base. This comment should be removed.